src/icon-layer.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  CompositeLayer,
+  CompositeLayerProps,
+  DefaultProps,
+  GetPickingInfoParams,
+  Layer,
+  LayersList,
+  assert,
+} from "@deck.gl/core";
+import { IconLayer } from "@deck.gl/layers";
+import type { IconLayerProps } from "@deck.gl/layers";
+import * as arrow from "apache-arrow";
+import * as ga from "@geoarrow/geoarrow-js";
+import {
+  assignAccessor,
+  extractAccessorsFromProps,
+  getGeometryVector,
+  invertOffsets,
+} from "./utils.js";
+import {
+  GeoArrowExtraPickingProps,
+  computeChunkOffsets,
+  getPickingInfo,
+} from "./picking.js";
+import { ColorAccessor, FloatAccessor, GeoArrowPickingInfo } from "./types.js";
+import { EXTENSION_NAME } from "./constants.js";
+import { validateAccessors } from "./validate.js";
+
+/** All properties supported by GeoArrowScatterplotLayer */
+export type GeoArrowIconLayerProps = Omit<
+  IconLayerProps<arrow.Table>,
+  | "data"
+  | "getPosition"
+  | "getRadius"
+  | "getFillColor"
+  | "getLineColor"
+  | "iconAtlas"
+  | "iconMapping"
```

Comment:
We only want to omit the properties that we redefine. It doesn't look like we redefine these, so you don't want to remove these from the type.